src/cim/components/ClusterDeployment/ClusterDeploymentHostsSelectionAdvanced.tsx

```diff
@@ -1,119 +1,53 @@
-  );
-
-  return (
-    <HostsTable
-      hosts={hosts}
-      content={content}
-      selectedIDs={selectedHostIdsField.value}
-      actionResolver={actionResolver}
-      className="agents-table"
-      {...actions}
-    >
-      <DefaultEmptyState
-        title="No hosts found"
-        content="No host matches provided labels/locations"
-      />
-    </HostsTable>
-  );
-};
+import AgentsSelectionTable from '../Agent/AgentsSelectionTable';
+import { Grid, GridItem } from '@patternfly/react-core';
 
 type ClusterDeploymentHostsSelectionAdvancedProps = {
   availableAgents: AgentK8sResource[];
 };
 
-const ClusterDeploymentHostsSelectionAdvanced: React.FC<ClusterDeploymentHostsSelectionAdvancedProps> = ({
-  availableAgents,
-}) => {
-  const { values } = useFormikContext<ClusterDeploymentHostsSelectionValues>();
+type FormValues = ClusterDeploymentHostsSelectionValues | ScaleUpFormValues;
 
-  const matchingAgents = React.useMemo(
-    () =>
-      availableAgents.filter((agent) => {
-        const labels = values.agentLabels.reduce((acc, curr) => {
-          const label = curr.split('=');
-          acc[label[0]] = label[1];
-          return acc;
-        }, {});
-        const matchesLocation = values.locations.length
-          ? values.locations.includes(
-              agent.metadata?.labels?.[AGENT_LOCATION_LABEL_KEY] || AGENT_NOLOCATION_VALUE,
-            )
-          : true;
-        const matchesLabels = agent.metadata?.labels
-          ? _.isMatch(agent.metadata?.labels, labels)
-          : true;
-        return matchesLocation && matchesLabels;
-      }),
-    [availableAgents, values.locations, values.agentLabels],
-  );
+const ClusterDeploymentHostsSelectionAdvanced = <T extends FormValues>({
+  availableAgents,
+}: ClusterDeploymentHostsSelectionAdvancedProps) => {
+  const { values } = useFormikContext<T>();
+
+  const matchingAgents = React.useMemo(() => {
+    return availableAgents.filter((agent) => {
+      const labels = values.agentLabels.reduce((acc, curr) => {
+        const label = curr.split('=');
+        acc[label[0]] = label[1];
+        return acc;
+      }, {});
+      const matchesLocation = values.locations.length
+        ? values.locations.includes(
+            agent.metadata?.labels?.[AGENT_LOCATION_LABEL_KEY] || AGENT_NOLOCATION_VALUE,
+          )
+        : true;
+      const matchesLabels = agent.metadata?.labels
+        ? _.isMatch(agent.metadata?.labels, labels)
+        : true;
+      return matchesLocation && matchesLabels;
+    });
+  }, [availableAgents, values.locations, values.agentLabels]);
 
   return (
     <>
-      <GridItem>
-        <LocationsSelector agents={availableAgents} />
-      </GridItem>
-      <GridItem>
-        <LabelsSelector agents={matchingAgents} />
-      </GridItem>
-      <GridItem>
-        <AgentsSelectionTable matchingAgents={matchingAgents} />
-      </GridItem>
+      <Grid hasGutter>
+        <GridItem span={12} lg={10} xl={9} xl2={7}>
```

Comment:
this might make fields too small in ACM create wizard, when YAML editor is open. Would be good to check.

Reply:
This component is not used in any of the views with yaml editor. I've fixed the vertical spacing of the fields.